pkg/identityserver/store/user_session_store.go

```diff
@@ -97,6 +98,30 @@ func (s *userSessionStore) GetSession(ctx context.Context, userIDs *ttnpb.UserId
 	return sessionProto, nil
 }
 
+func (s *userSessionStore) GetSessionByID(ctx context.Context, sessionID string) (*ttnpb.UserSession, error) {
+	defer trace.StartRegion(ctx, "get user session by session ID").End()
+	query := s.query(ctx, UserSession{}).Where(UserSession{Model: Model{ID: sessionID}})
+	var sessionModel UserSession
+	if err := query.Find(&sessionModel).Error; err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			return nil, errSessionNotFound.WithAttributes("session_id", sessionID)
+		}
+		return nil, err
+	}
+	query = s.query(ctx, Account{}).Where(Account{AccountID: sessionModel.UserID})
```

Comment:
The index is by `AccountID` and `AccountType`, so please query by both.
```suggestion
	query = s.query(ctx, Account{}).Where(Account{
		AccountID: sessionModel.UserID,
		AccountType: "user",
	})
```